### ISSUE TITLE
Changes to browser's session store for Piral debug-utils are ignored

### ISSUE THREAD

**Issue by Robbson** [smapiot/piral#590]
# Bug Report

Changes to browser's session store for Piral debug-utils are ignored when running the Piral instance (using `piral debug`) the first time or when opening it in a new browser tab.

## Prerequisites

- [x] Can you reproduce the problem in a [MWE](https://en.wikipedia.org/wiki/Minimal_working_example)?
- [x] Are you running the latest version?
- [x] Did you perform a search in the issues?

## Steps to Reproduce

1. Create a new Piral instance app using `piral new --target my-app`
2. Start the app using `npx piral debug` and choose _webpack5_ as bundler
3. Add `sessionStorage.setItem('dbg:view-state', 'off');` anywhere to the `index.tsx`
4. Open `http://localhost:1234` in a new browser tab
5. Open the browser's console and you will see the state debug output even though you have disabled it. This happens every time you start a new session (not using page refresh but opening a new browser tab)

## Expected behavior
My changes to the debug settings should always work, so there should be no state debug output in the console.

## Actual behavior
See step 5.

## Possible Origin/Solution
This behaviour has probably been introduced in Piral 0.14.30 when the Piral debug-utils got a new persistence feature (it worked just fine in 0.14.28).

**Comment by FlorianRappl** [comment]
I just tried this. It works. Sure you are doing the correct thing? All I did was paste

```js
sessionStorage.setItem('dbg:view-state', 'off')
```

and hit F5. No more view state output.

Please provide a MWE - otherwise I don't think this is a Piral issue, but rather something on your particular application or instance (maybe you override or change somehow `sessionStorage`?!).

**Comment by Robbson** [comment]
Hitting F5 doesn't clear the session store. You have to open it in a new tab.

**Comment by FlorianRappl** [comment]
> No, it's not, when you open it in a new tab (so using a new session store).

I think we have the problem here; you think the `sessionStorage` works across tabs. It does not. Each tab has a new session.

**Comment by Robbson** [comment]
Of course it is related. When you press F5, the session store (with the right value) still exist when the application gets reloaded. So Piral can pick it up in that case.

**Comment by Robbson** [comment]
> > No, it's not, when you open it in a new tab (so using a new session store).
> 
> I think we have the problem here; you think the `sessionStorage` works across tabs. It does not. Each tab has a new session. This has nothing to do with Piral - that's just how browsers work.

Where I did say that the `sessionStorage` works across tabs? That's confusing.

**Comment by FlorianRappl** [comment]
> Where I did say that the sessionStorage works across tabs? That's confusing.

You did above when you suggested that opening in a new tab should still take the value set in the `sessionStorage` closed tab.

Again, I tested it - it is picked up. There is no need to close / open a new tab. As mentioned, the browser opens a new `sessionStorage` per tab, so the previous value is of course not preserved. My test has shown that the value is picked up, so the reported issue is not an issue.

See also the video above.

**Comment by Robbson** [comment]
> You did above when you suggested that opening in a new tab should still take the value set in the `sessionStorage` closed tab.

No, I haven't at all. Please quote it!

And I have no idea why this should be the intended way as you said in your last sentence. Why should anywhone be bothered with state change debugging output when the state changes just work fine? 

The Piral-Debug-Utils seems to always get imported first, that's why `sessionStorage.setItem('dbg:view-state', 'off')` is just too late. And furthermore, it is still ignored from Piral because for some reason the storage [change listener](https://github.com/smapiot/piral/blob/dd15ddee171d01ade4f9f50938730abac69a80af/src/utilities/piral-debug-utils/src/debug.ts#L373) doesn't trigger. It only triggers when you make changes to the storage from the browser.

**Comment by FlorianRappl** [comment]
> The Piral-Debug-Utils seems to always get imported first, that's why sessionStorage.setItem('dbg:view-state', 'off') is just too late. And furthermore, it is still ignored from Piral because for some reason the storage [change listener](https://github.com/smapiot/piral/blob/dd15ddee171d01ade4f9f50938730abac69a80af/src/utilities/piral-debug-utils/src/debug.ts#L373) doesn't trigger. It only triggers when you make changes to the storage from the browser.

I mentioned what you need to do. You define the import order - make use of it.

> And I have no idea why this should be the intended way as you said in your last sentence. Why should anywhone be bothered with state change debugging output when the state changes just work fine?

I am not sure I get that sentence. The setting allows anyone to define (for the current session) to enable or disable state change output. If you don't want to see this then you can use the Piral Inspector and globally disable state change output. But still, this is (and should be) a personal preference. But then again, note that this was not the issue you raised, which is a bug - but instead, it would be a feature request to, e.g., change that default.

**Comment by Robbson** [comment]
Yes, a feature request sounds good to me. The defaults could be set using the CLI or the pilet.json or something like that?

Indeed, you could set the defaults in the Piral Inspector. But our Pilet developers usually don't use this browser extension because they spend most 95% of their time on components which are not related to Piral. When they start the Piral Emulator, the first thing they see is a pretty crowded console with hundred of internal state changes. And I would like to avoid that.
